Fix C# main screen plugin example and document C# `EditorInterface` singleton (#9399)
godotengine/godot-docs: tutorials/scripting/c_sharp/c_sharp_differences.rst

    Input.Singleton.JoyConnectionChanged += Input_JoyConnectionChanged;

> If you are developing main screen plugins using C#, it is essential to note the 
> distinctions between ``GDScript`` and ``C#`` during setup. Unlike in ``GDScript``, 
> this portion is not a static class in ``C#``. Consequently, you must employ the singleton 
> pattern to obtain an instance of the ``EditorInterface``:

Reviewer: ```suggestion
If you are developing main screen plugins, it is essential to note that ``EditorInterface``
is not a static class in C#, unlike in GDScript. Consequently, you must use the
singleton pattern to obtain an instance of the ``EditorInterface``:
```
